dbms/src/Storages/DeltaMerge/Index/MinMaxIndex.cpp

```diff
@@ -305,10 +305,10 @@ RSResults MinMaxIndex::checkNullableEqual(
             size_t pos = i * 2;
             size_t prev_offset = pos == 0 ? 0 : offsets[pos - 1];
             // todo use StringRef instead of String
-            auto min = String(chars[prev_offset], offsets[pos] - prev_offset - 1);
+            auto min = String(reinterpret_cast<const char *>(&chars[prev_offset]), offsets[pos] - prev_offset - 1);
```

Comment:
Add some description about why we **must** add this cast

Reply:
Added.